ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/searchtree/algorithms/SearchTree.java

```diff
@@ -370,7 +374,6 @@ private void optimizeLeaf(Leaf leaf) {
         } else {
             TECHNICAL_LOGS.info("No range actions to optimize");
         }
-        leaf.finalizeOptimization();
```

Comment:
why not keep this here to minimize the memory footprint of the search-tree, and only add it for lines 117 & 123?

Reply:
The networks are kept in memory anyways during the search tree by the network pool.
It's only important to make sure we remove the reference to the network when returning the leaf that will be stored in the results.

Reply:
(and as an example of the complexity, we'd also have to call it at line 345 if we don't call it at the top level, because of leaves that reach the stop criterion before optimization, with just the topo action applied)